src/Parsers/Access/ParserCreateUserQuery.cpp

```diff
@@ -120,7 +120,7 @@ namespace
                     return false;
                 value = ast->as<const ASTLiteral &>().value.safeGet<String>();
 
-                if (id_mode && expect_hash)
+                if (expect_hash)
```

Comment:
This condition is a bit inconsistent. Salt can be used only with `SHA256_PASSWORD`, but here we try to parse it for `DOUBLE_SHA1_HASH` too.

Reply:
Actually expect_hash will be set for SHA256_HASH, but let me check DOUBLE_SHA1_HASH...

Reply:
I don't see `expect_hash` will be set for `DOUBLE_SHA1_HASH` - in corresponding code `type` will be set in previous loop and code with setting `expect_hash` seems redundant to me...

Reply:
correction
now `SALT` can be used for both - SHA256_HASH and DOUBLE_SHA1_HASH with the same effect.

Reply:
I meant salt is not supported for DOUBLE_SHA1_HASH (and probably doesn't need to be supported) and better not to parse it here

Reply:
I would look at it other way around - currently salt is not added for `DOUBLE_SHA1_PASSWORD` - but maybe it should? and then salt for `DOUBLE_SHA1_HASH` will be more consistent. But anyway I see possibility to set salt for `DOUBLE_SHA1_HASH` as a feature.

Reply:
DOUBLE_SHA1 is often used with MySQL native protocol implementation which can't work with salt.
And if something is not going to work for me it's better to tell user about it.

Reply:
what the scenario of using of our DOUBLE_SHA1 with MySQL? Let's say we made `CREATE USER ... IDENTIFIED WITH double _sha1_**password**` - how this is used with MySQL?

Reply:
ClickHouse server can pretend to be MySQL server so if we enable that in the configuration any MySQL client can connect to ClickHouse server thinking it's a MySQL server via MySQL native protocol. For that connection one pass a user name and a password and they will be checked by ClickHouse if the corresponding user is defined `WITH double_sha1_password`

Reply:
Then I think there is no problem - salt is completely internal thing and transparent when password is used - salt is added to provided password internally, then this concatenation is hashed and compared with saved hash.
I can provide more information on why and how salt is used if you wish to make sure we are on the same page and I don't make any wrong conclusions...

Reply:
MySQL clients don't send plaintext password, they send hash instead. To apply salt we need plaintext password, we cannot apply salt to what already hashed.

Reply:
...of course I can remove it for consistency... what do you think will be the best?